Vaadin Flow Text Field is not null safe

_From @ugurdoksan on July 27, 2018 15:3_

When I call textField.setValue( null ); it throws "Caused by: java.lang.NullPointerException: Null value is not supported". It is not usefull...

_Copied from original issue: vaadin/vaadin-core#191_

--- a/src/main/java/com/vaadin/flow/component/textfield/PasswordField.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/PasswordField.java
@@ -22,6 +22,7 @@ import com.vaadin.flow.component.HasValidation;
 import com.vaadin.flow.component.InputNotifier;
 import com.vaadin.flow.component.KeyNotifier;
 import com.vaadin.flow.component.UI;
+import com.vaadin.flow.data.binder.Binder;
 import com.vaadin.flow.data.value.HasValueChangeMode;
 import com.vaadin.flow.data.value.ValueChangeMode;
 
@@ -353,6 +354,9 @@ public class PasswordField
      * Sets the value of this password field. If the new value is not equal to
      * {@code getValue()}, fires a value change event. Throws
      * {@code NullPointerException}, if the value is null.
+     * <p>
+     * Note: {@link Binder} will take care of the {@code null} conversion when
+     * integrates with password field, as long as no new converter is defined.
      * 
      * @param value
      *            the new value, not {@code null}

--- a/src/main/java/com/vaadin/flow/component/textfield/PasswordField.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/PasswordField.java
@@ -349,6 +349,30 @@ public class PasswordField
         return "";
     }
 
+    /**
+     * Sets the value of this password field. If the new value is not equal to
+     * {@code getValue()}, fires a value change event. Throws
+     * {@code NullPointerException}, if the value is null.
+     * 
+     * @param value
+     *            the new value
+     */
+    @Override
+    public void setValue(String value) {
+        super.setValue(value);
+    }
+
+    /**
+     * Returns the current value of the password field. By default, the empty
+     * password field will return an empty string.
+     * 
+     * @return the current value.
+     */
+    @Override
+    public String getValue() {
+        return super.getValue();
+    }
+
     @Override
     public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
         super.setRequiredIndicatorVisible(requiredIndicatorVisible);

--- a/src/main/java/com/vaadin/flow/component/textfield/PasswordField.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/PasswordField.java
@@ -355,7 +355,7 @@ public class PasswordField
      * {@code NullPointerException}, if the value is null.
      * 
      * @param value
-     *            the new value
+     *            the new value, not {@code null}
      */
     @Override
     public void setValue(String value) {

--- a/src/main/java/com/vaadin/flow/component/textfield/TextArea.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/TextArea.java
@@ -298,6 +298,30 @@ public class TextArea extends GeneratedVaadinTextArea<TextArea, String>
         return "";
     }
 
+    /**
+     * Sets the value of this text area. If the new value is not equal to
+     * {@code getValue()}, fires a value change event. Throws
+     * {@code NullPointerException}, if the value is null.
+     * 
+     * @param value
+     *            the new value
+     */
+    @Override
+    public void setValue(String value) {
+        super.setValue(value);
+    }
+
+    /**
+     * Returns the current value of the text area. By default, the empty text
+     * area will return an empty string.
+     * 
+     * @return the current value.
+     */
+    @Override
+    public String getValue() {
+        return super.getValue();
+    }
+
     @Override
     public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
         super.setRequiredIndicatorVisible(requiredIndicatorVisible);

--- a/src/main/java/com/vaadin/flow/component/textfield/TextArea.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/TextArea.java
@@ -21,6 +21,7 @@ import com.vaadin.flow.component.HasValidation;
 import com.vaadin.flow.component.InputNotifier;
 import com.vaadin.flow.component.KeyNotifier;
 import com.vaadin.flow.component.UI;
+import com.vaadin.flow.data.binder.Binder;
 import com.vaadin.flow.data.value.HasValueChangeMode;
 import com.vaadin.flow.data.value.ValueChangeMode;
 
@@ -302,6 +303,9 @@ public class TextArea extends GeneratedVaadinTextArea<TextArea, String>
      * Sets the value of this text area. If the new value is not equal to
      * {@code getValue()}, fires a value change event. Throws
      * {@code NullPointerException}, if the value is null.
+     * <p>
+     * Note: {@link Binder} will take care of the {@code null} conversion when
+     * integrates with text area, as long as no new converter is defined.
      * 
      * @param value
      *            the new value, not {@code null}

--- a/src/main/java/com/vaadin/flow/component/textfield/TextArea.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/TextArea.java
@@ -304,7 +304,7 @@ public class TextArea extends GeneratedVaadinTextArea<TextArea, String>
      * {@code NullPointerException}, if the value is null.
      * 
      * @param value
-     *            the new value
+     *            the new value, not {@code null}
      */
     @Override
     public void setValue(String value) {

--- a/src/main/java/com/vaadin/flow/component/textfield/TextField.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/TextField.java
@@ -343,7 +343,7 @@ public class TextField extends GeneratedVaadinTextField<TextField, String>
      * {@code NullPointerException}, if the value is null.
      * 
      * @param value
-     *            the new value
+     *            the new value, not {@code null}
      */
     @Override
     public void setValue(String value) {

--- a/src/main/java/com/vaadin/flow/component/textfield/TextField.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/TextField.java
@@ -21,6 +21,7 @@ import com.vaadin.flow.component.HasValidation;
 import com.vaadin.flow.component.InputNotifier;
 import com.vaadin.flow.component.KeyNotifier;
 import com.vaadin.flow.component.UI;
+import com.vaadin.flow.data.binder.Binder;
 import com.vaadin.flow.data.value.HasValueChangeMode;
 import com.vaadin.flow.data.value.ValueChangeMode;
 
@@ -342,7 +343,7 @@ public class TextField extends GeneratedVaadinTextField<TextField, String>
      * {@code getValue()}, fires a value change event. Throws
      * {@code NullPointerException}, if the value is null.
      * <p>
-     * Note: Binder will take care of the {@code null} conversion when
+     * Note: {@link Binder} will take care of the {@code null} conversion when
      * integrates with text field, as long as no new converter is defined.
      * 
      * @param value

--- a/src/main/java/com/vaadin/flow/component/textfield/TextField.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/TextField.java
@@ -337,6 +337,30 @@ public class TextField extends GeneratedVaadinTextField<TextField, String>
         return "";
     }
 
+    /**
+     * Sets the value of this text field. If the new value is not equal to
+     * {@code getValue()}, fires a value change event. Throws
+     * {@code NullPointerException}, if the value is null.
+     * 
+     * @param value
+     *            the new value
+     */
+    @Override
+    public void setValue(String value) {
+        super.setValue(value);
+    }
+
+    /**
+     * Returns the current value of the text field. By default, the empty text
+     * field will return an empty string.
+     * 
+     * @return the current value.
+     */
+    @Override
+    public String getValue() {
+        return super.getValue();
+    }
+
     @Override
     public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
         super.setRequiredIndicatorVisible(requiredIndicatorVisible);

--- a/src/main/java/com/vaadin/flow/component/textfield/TextField.java
+++ b/src/main/java/com/vaadin/flow/component/textfield/TextField.java
@@ -341,6 +341,9 @@ public class TextField extends GeneratedVaadinTextField<TextField, String>
      * Sets the value of this text field. If the new value is not equal to
      * {@code getValue()}, fires a value change event. Throws
      * {@code NullPointerException}, if the value is null.
+     * <p>
+     * Note: Binder will take care of the {@code null} conversion when
+     * integrates with text field, as long as no new converter is defined.
      * 
      * @param value
      *            the new value, not {@code null}

